### PR TITLE
feat: add release-notes target

### PR DESCRIPTION
**Pull request by vpavlin**
# Description
Adds a make target to generate release notes based on commit messages.

# Changes

- [x] new target added to Makefile

Example:

```
$ make release-notes 
## v0.17.0 (2023-05-16)

### Features

- add release-notes target ([ec9b222e](https://github.com/waku-org/nwaku/commit/ec9b222e))
- **cbindings:** first commit - waku relay ([#1632](https://github.com/waku-org/nwaku/issues/1632)) ([#1714](https://github.com/waku-org/nwaku/issues/1714)) ([2defbd23](https://github.com/waku-org/nwaku/commit/2defbd23))
- example using filter and lightpush ([#1720](https://github.com/waku-org/nwaku/issues/1720)) ([8987d4a3](https://github.com/waku-org/nwaku/commit/8987d4a3))
- configure protected topics via cli ([#1696](https://github.com/waku-org/nwaku/issues/1696)) ([16b44523](https://github.com/waku-org/nwaku/commit/16b44523))
- **mem-analysis:** Adding Dockerfile_with_heaptrack ([#1681](https://github.com/waku-org/nwaku/issues/1681)) ([9b9172ab](https://github.com/waku-org/nwaku/commit/9b9172ab))
- add metrics with msg size histogram ([#1697](https://github.com/waku-org/nwaku/issues/1697)) ([67e96ba8](https://github.com/waku-org/nwaku/commit/67e96ba8))
...
```

$ make release-notes 
### Features

- add release-notes target ([ec9b222e](https://github.com/waku-org/nwaku/commit/ec9b222e))
- **cbindings:** first commit - waku relay ([#1632](https://github.com/waku-org/nwaku/issues/1632)) ([#1714](https://github.com/waku-org/nwaku/issues/1714)) ([2defbd23](https://github.com/waku-org/nwaku/commit/2defbd23))
- example using filter and lightpush ([#1720](https://github.com/waku-org/nwaku/issues/1720)) ([8987d4a3](https://github.com/waku-org/nwaku/commit/8987d4a3))
- configure protected topics via cli ([#1696](https://github.com/waku-org/nwaku/issues/1696)) ([16b44523](https://github.com/waku-org/nwaku/commit/16b44523))
- **mem-analysis:** Adding Dockerfile_with_heaptrack ([#1681](https://github.com/waku-org/nwaku/issues/1681)) ([9b9172ab](https://github.com/waku-org/nwaku/commit/9b9172ab))
- add metrics with msg size histogram ([#1697](https://github.com/waku-org/nwaku/issues/1697)) ([67e96ba8](https://github.com/waku-org/nwaku/commit/67e96ba8))
...

### CODE REVIEW

**Comment by vpavlin** [comment]
I will remove this target from #1712. That one is not gonna get merge before release, but would like to use the release notes for 0.17.0, hence this separate PR

**Review by alrevuelta — APPROVED** [review]
I usually do it "manually" with `Generate release notes` which does a great job including links to PRs, description etc. Outputs that are read by humans shouldn't imho be fully automated, but just leverage technology like "Generate release notes" does, which allows for some human corrections.

I believe @jm-clius also prefered it to be a bit more manual? To do some last minute corrections?

Not a blocker though, just an opinion from someone not in charge of the releases.

image

**Comment by vpavlin** [comment]
Yeah, it will stay manual for now. My plan here is to have a branch and tag with release candidate builds (binaries, container images..) prepped automatically, then have a manual step to review, edit and push final release/tag, but ideally the initial steps of release prep should be automatic.

File: Makefile [hunk 1]
```diff
@@ -308,3 +308,19 @@ cwaku_example: | build cbindings
...
+###################
+
+release-notes:
+	docker run \
+		-it \
+		--rm \
+		-v $${PWD}:/opt/sv4git/repo:z \
+		-u $(shell id -u) \
+		quay.io/vpavlin0/sv4git:latest \
+			release-notes |\
```
Comment (Ivansete-status) [inline Makefile:323]: Shall we have a `waku` account instead?

**Comment by vpavlin** [comment]
@alrevuelta I tried with the Generate release notes and it does not look very good with semantic commits:

Screenshot from 2023-05-16 14-21-21

vs. sv4git
Screenshot from 2023-05-16 14-23-01

We can change the format if we need to and still manually edit things as much as we want:) But I personally like the format way more than what `Generate release notes` produces automatically

**Review by rymnc — APPROVED** [review]
LGTM, thanks for this

**Review by jm-clius — APPROVED** [review]
LGTM. I previously used the `Generate release notes...` but this required a significant amount of manual rewriting. This seems better and more configurable to make the manual intervention a bit less tedious. Good work. :)

**Comment by alrevuelta** [comment]
@vpavlin thanks, indeed looks better than the github release notes.

File: Makefile (hunk 1 above)
Comment (vpavlin) [reply Makefile]: @Ivansete-status fixed

Comment (Ivansete-status) [reply Makefile]: Thanks @vpavlin !

**Review by Ivansete-status — APPROVED** [review]
That PR is awesome. Thanks for it!

**Pull request merged**